DFE-Digital/sts-plan-technology-for-your-school
# refactor: Refactor model configuring for CmsDbContext

## Overview

This PR cleans up the configuration of the `CmsDbContext`. It moves all the different entity type configurations out of the `OnModelConfiguring` method, by creating an `IEntityTypeConfiguration` for each one. We already had some of these, particularly the more complex configurations, but it was inconsistent. It's now consistent, and the `CmsDbContext` looks much cleaner.

## Changes

### Minor

- Move _all_ entity type configuration in `CmsDbContext` to their own concrete classes
- Other very minor changes, mostly things like ordering methods by access level, refactoring the `GetService` calls in the `CmsDbContext`, etc. 

## How to review the PR

1. Run the web app
2. Ensure nothing is broken
3. Success!

Would be good to test `Page` content changes from Contentful, i.e. CRUD operations on a content that is a `Page` content type. The change I made is _minor_ and _should not impact anything_ and _seems to work fine_ but 🤷 

## Checklist

Delete any rows that do not apply to the PR.
